### PR TITLE
refactor(router): rename "as" to "name" in RouteConfig

### LINKED ISSUE #4622: refactor(router): rename "as" to "name" in RouteConfig
John Papa proposes we rename `as` to `name` in `RouteConfig` to reduce some of the confusion about its purpose:

So this:

```
@RouteConfig([
  { path: '/foo', component: MyCmp, as: 'MyCmp' }
])
```

Would become this:

```
@RouteConfig([
  { path: '/foo', component: MyCmp, name: 'MyCmp' }
])
```

/cc @johnpapa

### PR DESCRIPTION
**Pull request by vamsivarikuti**
Closes #4622

### CODE REVIEW

**Comment by vamsivarikuti** [comment]
@btford This is my first contribution. I'm not sure why is this build breaking. Build successful in my local, I have done `gulp check-format`, `gulp build` it was successful. When i do `npm test` it says `Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL`

Any suggestions?

File: modules/angular2/src/router/route_config_nomalizer.ts [hunk 1]
```diff
@@ -17,8 +17,14 @@ export function normalizeRouteConfig(config: RouteDefinition): RouteDefinition {
     throw new BaseException(
         `Route config should contain exactly one "component", "loader", or "redirectTo" property.`);
   }
+  if ((+!!config.as) + (+!!config.name) != 1) {
```
Comment (mhevery) [inline modules/angular2/src/router/route_config_nomalizer.ts]: this syntax is not allowed in Dart. Please rewrite in less clever way.

**Comment by mhevery** [comment]
Looks like your change broke existing tests. See: https://travis-ci.org/angular/angular/jobs/87315369#L287

File: modules/angular2/src/router/route_config_nomalizer.ts (hunk 1 above)
Comment (vamsivarikuti) [inline modules/angular2/src/router/route_config_nomalizer.ts]: @mhevery I tried to follow same syntax as above line 16, shall i change that too?

**Comment by btford** [comment]
@vamsivarikuti – no worries. I can help with the Dart failures.

**Comment by vamsivarikuti** [comment]
@btford -- So, What's the status of this? Need any changes? looks like there are some conflicts.

**Comment by btford** [comment]
I'll handle the conflicts. Thanks for your help!

**Comment by vamsivarikuti** [comment]
I should be thanking you guys for everything you have done for Angular. Angular is awesome and it keeps food on my plate. I never miss any of your talks (on youtube obviously :smile: ), waiting for V2.. 

Thank you guys.

**Pull request closed without merging**